java/build.gradle

```diff
@@ -98,6 +99,11 @@ if (cmakeBuildDir != null) {
 
 	// Overwrite jar location
 	task allJar(type: Jar) {
+		manifest {
+			attributes('Automatic-Module-Name': project.group,
+					'Implementation-Title': 'onnxruntime',
+					'Implementation-Version': project.version)
+		}
 		into("META-INF/maven/$project.group/$project.name") {
```

Comment:
This line isn't right, it should be `$artifactId` not `$project.name`. That's a bug on my part as I don't usually adjust the artifactId, it's always set to the project name, so when we change the artifactId we should make sure it's put in the right place.

Reply:
This is really bad because it will take may be yet another week of testing for the pipeline.

---
In reply to: [436887785](https://github.com/microsoft/onnxruntime/pull/4159#discussion_r436887785) [](ancestors = 436887785)

Reply:
It's only the pom. If it's not in the right place then some automatic maven based tooling might not extract the pom correctly (e.g. artifactory).

Reply:
It doesn't affect the code or anything at runtime, it's purely a tooling convenience.

Reply:
It affects the pipeline.

---
In reply to: [436977216](https://github.com/microsoft/onnxruntime/pull/4159#discussion_r436977216) [](ancestors = 436977216)

Reply:
So if we change only this line then sources and javadoc POM would remain at the old path. If I did change those too it results in an error:

 A problem occurred evaluating root project 'onnxruntime'.
  > Could not get unknown property 'artifactId' for task ':sourcesJar' of type org.gradle.api.tasks.bundling.Jar.

How important for us to be consistent and not incur the ire of the community?

---
In reply to: [436976973](https://github.com/microsoft/onnxruntime/pull/4159#discussion_r436976973) [](ancestors = 436976973)

Reply:
Actually, this errors out even for allJar only:
 A problem occurred evaluating root project 'onnxruntime'.
  > Could not get unknown property 'artifactId' for task ':allJar' of type org.gradle.api.tasks.bundling.Jar.


---
In reply to: [437019595](https://github.com/microsoft/onnxruntime/pull/4159#discussion_r437019595) [](ancestors = 437019595,436976973)